src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -49,9 +48,15 @@ trait RegularSync {
       waitingForActor = None
       handleBlockBodies(peer, blockBodies)
 
-    case block: BroadcastBlocks if broadcasting =>
+    case BroadcastBlocks(newBlocks) if broadcasting =>
       //FIXME: Decide block propagation algorithm (for now we send block to every peer) [EC-87]
-      peersToDownloadFrom.keys.foreach(_.ref ! block)
+      val blocksToSendToEachPeer = for {
+        peer <- peersToDownloadFrom.keys
```

Comment:
That's one possible solution. Another one (and I think I like it slightly more) is to add possibility to send messages to peers in peerEventBus, so that we can do, for example: `peerEventBus ! MessageToPeers(block, PeerSelector.AllPeers)` or `MessageToPeers(block, PeerSelector.Handshaked)` etc